File: ros2/cpp_topic/src/subscriber.cpp
#include "my_subscriber.hpp"

using namespace std::placeholders;

SubNode::SubNode(): Node("subscriber_node"){
  RCLCPP_DEBUG(this->get_logger(),"Subscriber has been started.");
  subs_int    = this->create_subscription<int16Msg>("cpp_topic_int", 10,
                std::bind(&SubNode::subsCallbackInt, this, _1));
  subs_string = this->create_subscription<stringMsg>("cpp_topic_string", 10,
                std::bind(&SubNode::subsCallbackString, this, _1));
}

SubNode::~SubNode(){
  RCLCPP_DEBUG(this->get_logger(), "Subscriber terminated...");
}

void SubNode::subsCallbackInt(const int16Msg msg){
  RCLCPP_INFO(this->get_logger(),"int: %d",msg.data);
}

void SubNode::subsCallbackString(const stringMsg msg){
  RCLCPP_INFO(this->get_logger(),"string: %s",msg.data.c_str());
}

int main(int argc, char ** args){
    rclcpp::init(argc, args);
    rclcpp::spin(std::make_shared<SubNode>());
    rclcpp::shutdown();
}

File: ros2/py_topic/py_topic/subscriber.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class MyNode(Node):
    def __init__(self):
        super().__init__("py_listener_node")
        self.subscriber_ = self.create_subscription(
            String, "py_test_counter", self.subscriberCallback, 10)
        self.get_logger().info("py_listener started...")

    def subscriberCallback(self, msg):
        self.get_logger().info(msg.data)


def main(args=None):
    rclpy.init(args=args)
    rclpy.spin(MyNode())
    rclpy.shutdown()


if __name__ == "__main__":
    main()